### DISCUSSION TITLE
Duplicate Records within Milliseconds Of Each Other (Rare Occasions)

### DISCUSSION THREAD

**Discussion by kiwavi** [prisma/prisma#23474] [Q&A] [answered]
### Question

Hello, I've experienced a situation whereby I'm getting duplicate records that are witihin miliiseconds of each other on the createdAt field which is set to default to the current now. These records are usually generated via a prisma create function on a postgres database. This happens rarely as noted on title. Thanks

The latest example is `2024-03-13 05:48:39.199` and `2024-03-13 05:48:39.2`

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nikolasburk** [comment]
Hey @kiwavi 👋 thanks for raising this!

Can you share some more information about your project, e.g. what stack you use? Also it would be useful to know when and where the `create` query that produces duplicate records is executed. Is there a chance it may get executed twice? Does this happen for all `create` queries in your app or just for a specific one?

The more information you can share that would help us pin down and reproduce the issue, the better 🙂 Ideally you could even share a repo with a minimal reproduction, that we can take a look at? 🙏

**Reply by kiwavi** [reply]
Basically, I receive some data and insert a record into a table with the value 'pending' in a field called withdrawn - as long as they do not have another record with 'pending'. If the same person posts some data again, I check if they have a record that still has the value 'pending' in the withdrawn field. If true, I raise an error. Therefore, a record can only be created when the person requesting does not have another record with 'pending'. Therefore sending the same data twice (even immediately) fails on the second request since, on the first request, a record will be instantiated with 'pending' on the withdrawn field. In this case however, the second one still went through, and the createdat timestamp is milliseconds apart. 

```
Using prisma 4.12
Backend: ExpressJS
DB: PostgreSQL 12.18 (Ubuntu 12.18-0ubuntu0.20.04.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0, 64-bit
```

Here's the code. (some unnecessary parts ommitted)

```

// request withdrawal by seller
router.post("/request", async (req, res) => {
    // check for pending withdrawals
    let lastWithdrawal = await prisma.withdrawals.findFirst({
        where: {
           userid: user.id,
        },
        orderBy: {createdAt: 'desc'}
    });

    if (lastWithdrawal && lastWithdrawal.status === 'pending') {
        return res.status(400).json({success: false, message: "An pending withdrawal request already exists"});
    }

    try {
        let withdrawal = await prisma.withdrawals.create({
            data: {                
                users: {connect: {id: parseInt(req.user?._id)}},
                packages: packages,
                status: "pending",
            }
        });

            return res.status(200).json({success: true, message: 'withdrawal request successful'});
        } else {
            return res.status(400).json({
                success: false,
                message: `withdrawal request failed`
            });
        }

    } catch (error) {
        logError(req.url, 500, error);
        return res.status(500).json({success: false, message: 'something went wrong.'})
    }
});
```

**Reply by kiwavi** [reply, marked as answer]
For some reason this issue stopped when I stopped using connect while creating the particular record. I am yet to investigate why that one-line change fixed it.
